macOS/DuckDuckGo/NetworkProtection/NetworkExtensionTargets/NetworkExtensionTargets/MacPacketTunnelProvider.swift

```diff
@@ -535,24 +535,98 @@ final class MacPacketTunnelProvider: PacketTunnelProvider {
         Logger.networkProtectionMemory.log("[-] MacPacketTunnelProvider")
     }
 
-    // MARK: - NEPacketTunnelProvider
-
     public override func load(options: StartupOptions) async throws {
         try await super.load(options: options)
-
-#if NETP_SYSTEM_EXTENSION
-        loadExcludeLocalNetworks(from: options)
-#endif
+        
+        // macOS-specific options
+        try loadVPNSettings(from: options)
+        loadAuthVersion(from: options)
+        if !Self.isUsingAuthV2 {
+            try await loadAuthToken(from: options)
+        } else {
+            try await loadTokenContainer(from: options)
+        }
     }
 
-    private func loadExcludeLocalNetworks(from options: StartupOptions) {
-        switch options.excludeLocalNetworks {
-        case .set(let exclude):
-            settings.excludeLocalNetworks = exclude
+    private func loadVPNSettings(from options: StartupOptions) throws {
+        switch options.vpnSettings {
+        case .set(let settingsSnapshot):
+            settingsSnapshot.applyTo(settings)
         case .useExisting:
             break
         case .reset:
-            settings.excludeLocalNetworks = true
+            // VPN settings are required - if we're in reset case, it means they were missing or invalid
+            throw TunnelError.settingsMissing
+        }
+    }
+
+    private func loadAuthVersion(from options: StartupOptions) {
+        switch options.isAuthV2Enabled {
+        case .set(let newAuthVersion):
+            Logger.networkProtection.log("Set new isAuthV2Enabled")
+            Self.isUsingAuthV2 = newAuthVersion
+        case .useExisting:
+            Logger.networkProtection.log("Use existing isAuthV2Enabled")
+        case .reset:
+            Logger.networkProtection.log("Reset isAuthV2Enabled")
+        }
+        Logger.networkProtection.log("Load isAuthV2Enabled: \(Self.isUsingAuthV2, privacy: .public)")
+    }
+
+    private func loadAuthToken(from options: StartupOptions) async throws {
+        let tokenHandler = tokenHandlerProvider()
```

Comment:
I'm also seeing the same issue locally as reported by the check failures below, looks like `tokenHandlerProvider()` isn't accessible.

Reply:
Sorry about this, will look into it tomorrow.